anchor-spl: `Hash` is missing the `new_from_array` method

Getting the following error when using any dependency, which pulls anchor-syn, inside a Solana program. This issue occurs only in `bpf` target.

```
error[E0599]: no function or associated item named `new_from_array` found for struct `Hash` in the current scope
   --> src/hash.rs:110:15
    |
12  | pub struct Hash(pub [u8; HASH_BYTES]);
    | --------------- function or associated item `new_from_array` not found for this struct
...
110 |         Hash::new_from_array(hash_result)
    |               ^^^^^^^^^^^^^^ function or associated item not found in `Hash`

   Compiling semver v0.11.0
For more information about this error, try `rustc --explain E0599`.
error: could not compile `anchor-syn` due to previous error
warning: build failed, waiting for other jobs to finish...
```

The method is indeed not there.

https://github.com/coral-xyz/anchor/blob/16c998ab436ba3f61d0778d6033f906585461f53/lang/syn/src/hash.rs#L76-L84

But it's being called here:

https://github.com/coral-xyz/anchor/blob/16c998ab436ba3f61d0778d6033f906585461f53/lang/syn/src/hash.rs#L110

--- a/lang/syn/src/hash.rs
+++ b/lang/syn/src/hash.rs
@@ -78,6 +78,10 @@ impl Hash {
         Hash(<[u8; HASH_BYTES]>::try_from(hash_slice).unwrap())
     }
 
+    pub const fn new_from_array(hash_array: [u8; HASH_BYTES]) -> Self {
+        Self(hash_array)
+    }
+
     pub fn to_bytes(self) -> [u8; HASH_BYTES] {
         self.0
     }

